docs/source/bft_configuration.md

```diff
@@ -0,0 +1,228 @@
+
+This is useful for cases where you want TLS certificates issued by the
+organizational CAs, but used only by the cluster nodes to communicate among each
+other, and TLS certificates issued by a public TLS CA for the client facing API.
+
+* `ClientCertificate`, `ClientPrivateKey`: The file path of the client TLS certificate
+  and corresponding private key.
+* `ListenPort`: The port the cluster listens on.
+  It must be same as `consenters[i].Port` in Channel configuration.
+  If blank, the port is the same port as the orderer general port (`general.listenPort`)
+* `ListenAddress`: The address the cluster service is listening on.
+* `ServerCertificate`, `ServerPrivateKey`: The TLS server certificate key pair
+  which is used when the cluster service is running on a separate gRPC server
+  (different port).
+
+Note: `ListenPort`, `ListenAddress`, `ServerCertificate`, `ServerPrivateKey` must
+be either set together or unset together.
+If they are unset, they are inherited from the general TLS section,
+in example `general.tls.{privateKey, certificate}`.
+When general TLS is disabled:
+- Use a different `ListenPort` than the orderer general port
+- Properly configure TLS root CAs in the channel configuration.
+
+There are also hidden configuration parameters for `general.cluster` which can be
+used to further fine tune the cluster communication or replication mechanisms:
+
+* `SendBufferSize`: Regulates the number of messages in the egress buffer.
+* `DialTimeout`, `RPCTimeout`: Specify the timeouts of creating connections and
+  establishing streams.
+* `ReplicationBufferSize`: the maximum number of bytes that can be allocated
+  for each in-memory buffer used for block replication from other cluster nodes.
+  Each channel has its own memory buffer. Defaults to `20971520` which is `20MB`.
+* `PullTimeout`: the maximum duration the ordering node will wait for a block
+  to be received before it aborts. Defaults to five seconds.
+* `ReplicationRetryTimeout`: The maximum duration the ordering node will wait
+  between two consecutive attempts. Defaults to five seconds.
+* `TLSHandshakeTimeShift`: If the TLS certificates of the ordering nodes
+  expire and are not replaced in time (see TLS certificate rotation below),
+  communication between them cannot be established, and it will be impossible
+  to send new transactions to the ordering service.
+  To recover from such a scenario, it is possible to make TLS handshakes
+  between ordering nodes consider the time to be shifted backwards a given
+  amount that is configured to `TLSHandshakeTimeShift`.
+  This setting only applies when a separate cluster listener is in use.  If
+  the cluster service is sharing the orderer's main gRPC server, then instead
+  specify `TLSHandshakeTimeShift` in the `General.TLS` section.
+
+**Consensus parameters:**
+
+* `WALDir`: the location at which Write Ahead Logs for BFT are stored.
+  Each channel will have its own subdirectory named after the channel ID.
+
```

Comment:
Just to confirm, the Raft SnapDir, EvictionSuspicion, and TickIntervalOverride don't apply to BFT?

Reply:
correct. SmartBFT has no snap dir, since it piggybacks snapshots on Fabric blocks, because in SmartBFT, the WAL may garbage collect itself every Fabric block.